meilisearch-types/src/keys.rs

```diff
@@ -179,193 +180,271 @@ fn parse_expiration_date(
+    pub const DUMPS_CREATE: u32 = A::DumpsCreate.bits();
+    pub const DUMPS_ALL: u32 = A::DumpsAll.bits();
+    pub const SNAPSHOTS_CREATE: u32 = A::SnapshotsCreate.bits();
+    pub const VERSION: u32 = A::Version.bits();
+    pub const KEYS_CREATE: u32 = A::KeysAdd.bits();
+    pub const KEYS_GET: u32 = A::KeysGet.bits();
+    pub const KEYS_UPDATE: u32 = A::KeysUpdate.bits();
+    pub const KEYS_DELETE: u32 = A::KeysDelete.bits();
+    pub const EXPERIMENTAL_FEATURES_GET: u32 = A::ExperimentalFeaturesGet.bits();
+    pub const EXPERIMENTAL_FEATURES_UPDATE: u32 = A::ExperimentalFeaturesUpdate.bits();
+}
+
+impl<E: DeserializeError> Deserr<E> for Action {
+    fn deserialize_from_value<V: deserr::IntoValue>(
+        value: deserr::Value<V>,
+        location: deserr::ValuePointerRef<'_>,
+    ) -> Result<Self, E> {
+        match value {
+            deserr::Value::String(s) => match Self::get_action(&s) {
+                Some(action) => Ok(action),
+                None => Err(deserr::take_cf_content(E::error::<std::convert::Infallible>(
+                    None,
+                    deserr::ErrorKind::UnknownValue {
+                        value: &s,
+                        accepted: &Self::SERDE_MAP_ARR.map(|(ser_action, _)| ser_action),
+                    },
+                    location,
+                ))),
+            },
+            _ => Err(take_cf_content(E::error(
+                None,
+                deserr::ErrorKind::IncorrectValueKind {
+                    actual: value,
+                    accepted: &[deserr::ValueKind::String],
+                },
+                location,
+            ))),
+        }
+    }
+}
+
+impl Serialize for Action {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(Self::get_action_serde_name(self))
+    }
+}
+
+impl<'de> Deserialize<'de> for Action {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct Visitor;
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = Action;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                write!(formatter, "the name of a valid action (string)")
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                match Self::Value::get_action(s) {
+                    Some(action) => Ok(action),
+                    None => Err(E::invalid_value(serde::de::Unexpected::Str(s), &"a valid action")),
+                }
+            }
+        }
+
+        deserializer.deserialize_str(Visitor)
+    }
+}
+
+impl Sequence for Action {
+    const CARDINALITY: usize = Self::SERDE_MAP_ARR.len();
+
+    fn next(&self) -> Option<Self> {
+        let mut iter = Self::FLAGS.iter();
```

Comment:
Possibly a linked list solution would be better implemented on private properties of `self`, but it's probably not worth it.

Reply:
As long as `Action` flags are ordered ascending by value manually, we're more efficient the current way.